tools/small_rna_maps/small_rna_maps.py

```diff
@@ -190,14 +191,23 @@ def compute_coverage(self, map_dictionary, out, quality=10):
                     if not map_dictionary[(chrom, pos+1, 'F')]:
                         map_dictionary[(chrom, pos+1, 'F')] = []
         for key in map_dictionary:
-            coverage = self.bam_object.count_coverage(
-                                                reference=key[0],
-                                                start=key[1]-1,
-                                                end=key[1],
-                                                quality_threshold=quality)
-            """ Add the 4 coverage values """
-            coverage = [sum(x) for x in zip(*coverage)]
-            coverage_dictionary[key] = coverage[0]
+            try:
+                coverage = self.bam_object.count_coverage(
+                                                    reference=key[0],
+                                                    start=key[1]-1,
+                                                    end=key[1],
+                                                    quality_threshold=quality)
+                """ Add the 4 coverage values """
+                coverage = [sum(x) for x in zip(*coverage)]
+                coverage_dictionary[key] = coverage[0]
+            except TypeError:
+                if key[0] not in missing:
+                    missing[key[0]] = [key[1]-1]
+                else:
+                    missing[key[0]].append(key[1])
+        if len(missing) > 0:
+            for chrom in sorted(missing):
+                 coverage_dictionary[chrom] = [0]*(missing[chrom][1] - missing[chrom][0]+1)
```

Comment:
@ccedmendoza could you elaborate a little bit on the missing dictionary and its purpose ?

Reply:
@drosofff the `key` in `for key in map_dictionary:` contains every chromosome of the reference. Which is normal since we want to output empty graphs if the chromosome wasn't mapped.
The problem is that we can't use the   `count_coverage` if there is no reads (no positions in map_dicrionary for this chromosome). So the missing dictionary is here to handle unaligned chromosomes. By assigning the value 0 to all their positions.